pkg/vmprovider/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -157,6 +157,9 @@ func vmTests() {
 					},
 					AddressType: string(types.VirtualEthernetCardMacTypeGenerated),
 					MacAddress:  "00:0c:29:93:d7:27",
+					ResourceAllocation: &types.VirtualEthernetCardResourceAllocation{
```

Comment:
This is in the test, but are we copying it from the config spec in the implementation?

Reply:
I don't fully follow your question. But here `ethCard` is later serialized into the class ConfigSpec, and later in the implementation, `networkDevices[]` is the devices from that ConfigSpec, in which some fields - like the backing, and now the MAC address from NCP - are then overwritten. NCP/NetOP don't have a `ResourceAllocation` or `WOL` so whatever is set in the class, if anything, is just blindly used.

Reply:
Ahhh. I see what you mean. I had the relationship backwards. I thought this was copying the info FROM the ConfigSpec INTO the NIC list we already created. Ack. Makes sense. Approved.